src/diffusers/models/attention_processor.py

```diff
@@ -2296,22 +2337,27 @@ def __call__(
         hidden_states = hidden_states.to(query.dtype)
 
         # for ip-adapter
-        ip_key = self.to_k_ip(ip_hidden_states)
```

Comment:
`ip_hidden_states` shape is (batch_size, num_tokens, embedding_dim). If you iterate over it it you get `current_ip_hidden_states` whose shape is (num_tokens, embedding_dim). Therefore:

> `ip_key = to_k_ip(current_ip_hidden_states)`
> `ip_key = ip_key.view(**batch_size**, -1, attn.heads, head_dim).transpose(1, 2)`

makes no sense. Is `ip_hidden_states` (ip_module, batch, num_tokens, embed_dim) instead of (batch, num_tokens, embed_dim)?